src/com/okta/SmsAuthenticator.java

```diff
@@ -0,0 +1,37 @@
+package com.okta;
+
+import net.jradius.client.RadiusClient;
+import net.jradius.client.auth.PAPAuthenticator;
+import net.jradius.dictionary.Attr_UserPassword;
+import net.jradius.exception.RadiusException;
+import net.jradius.packet.RadiusPacket;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+public class SmsAuthenticator extends PAPAuthenticator {
+    private RadiusClient radiusClient;
+
+    public SmsAuthenticator(RadiusClient rc) {
+        radiusClient = rc;
+    }
+
+    @Override
+    public void processChallenge(RadiusPacket request, RadiusPacket challenge) throws RadiusException {
```

Comment:
Why is this separation needed for each factor? The code currently supports all 2nd factor flow - there is no need to write extra code for each factor type.